src/main/scala/net/ceedubs/ficus/readers/ArbitraryTypeReader.scala

```diff
@@ -11,22 +12,48 @@ trait ArbitraryTypeReader {
 
 object ArbitraryTypeReader extends ArbitraryTypeReader
 
+trait HyphenCaseArbitraryTypeReader {
+  implicit def arbitraryTypeValueReader[T]: ValueReader[T] = macro ArbitraryTypeReaderMacros.arbitraryTypeValueReaderWithHyphenCase[T]
+}
+
+object HyphenCaseArbitraryTypeReader extends HyphenCaseArbitraryTypeReader
+
 object ArbitraryTypeReaderMacros {
   import scala.reflect.macros.blackbox.Context
 
-  def arbitraryTypeValueReader[T : c.WeakTypeTag](c: Context): c.Expr[ValueReader[T]] = {
+  trait NameMapper {
+    def map(name: String): String
+  }
+
+  object defaultMapper extends NameMapper {
+    def map(name: String) = name
+  }
+
+  object hyphenCaseMapper extends NameMapper {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Object name does not match the regular expression '^[A-Z][A-Za-z0-9]*$'](https://www.codacy.com/app/kailuo-wang/ficus/file/1805786830/issues/source?bid=3018830&fileBranchId=3073710#l32)